### PR TITLE
profile indicator subcategory dropdown filtering

### PR DESCRIPTION
**Pull request by RomanMelnyk113**
## Description
When creating a new ProfileIndicator, the subcategory dropdown does not filter options related to the selected profile - currently, subcategories from all profiles are display

## Related Issue
https://github.com/OpenUpSA/wazimap-ng/issues/157

## How to test it locally
Pre-conditions:

- different profiles available
- multiple subcategories assigned to different profiles (through categories)

To test:
1. Open "Add profile indicator" page (http://localhost:8000/admin/profile/profileindicator/add/)
2. Select different profiles from the dropdown list
3. Check available subcategories for each profile

### Added
1. New endpoint to get the list of all subcategories related to specific profile - `/api/v1/profiles/{id}/subcategories/`
2. `profile-indicator-admin.js` file with logic to fetch subcategories on profile select

## Checklist

- [x]  🚀 is the code ready to be merged and go live?
- [x]  🛠 does it work (build) locally

### Pull Request

- [x]  📰 good title
- [x]  📝good description
- [x]  🔖 issue linked
- [ ]  📖 changelog filled out

### Commits

- [x]  commits are clean
- [x]  commit messages are clean

### Code Quality

- [x]  🚧 no commented out code
- [x]  🖨 no unnecessary logging
- [x]  🎱 no magic numbers
- [ ]  black was run locally (as part of the pre-commit hook)

### Testing

- [ ]  ✅ added (appropriate) unit tests
- [ ]  💢 edge cases in tests were considered
- [x]  ✅ ran tests locally & are passing

### CODE REVIEW

File: wazimap_ng/general/static/js/profile-indicator-admin.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        }
+
+        function loadSubcategory(selectedProfile) {
+            if (selectedProfile) {
+                var url = `/api/v1/profiles/${selectedProfile}/subcategories/`;
+                $.ajax({
+                    url: url,
+                    success: function (data) {
+                        clearSubcategories();
+                        $.each(data, function (key, value) {
```
[the remark this answers is not in the document]
Comment (RomanMelnyk113) [reply wazimap_ng/general/static/js/profile-indicator-admin.js:28]: clearSubcategories(); is used twice to avoid "select" input flickering which appears during AJAX call

**Comment by RomanMelnyk113** [comment]
@milafrerichs do we need to write tests here? because I don't see any tests for the admin side

**Review by milafrerichs — CHANGES_REQUESTED** [review]
was meaning to submit this last week but did not complete it. sorry for the delay.

File: wazimap_ng/general/static/js/profile-indicator-admin.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+            }
+        });
+
+        function clearSubcategories() {
+            $subcategoryEl.empty();
+            $subcategoryEl.append($("<option></option>").attr("value", "").text("----------"));
+        }
+
+        function loadSubcategory(selectedProfile) {
+            console.log("selectedProfile", selectedProfile)
```
[the remark this answers is not in the document]
Comment (RomanMelnyk113) [reply wazimap_ng/general/static/js/profile-indicator-admin.js]: wow, sorry, missed that one...
removed.

File: wazimap_ng/general/static/js/profile-indicator-admin.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+            // trigger "loadSubcategory" func only when it was manually changed
+            if (e.originalEvent !== undefined) {
+                // load subcategories based on selected profile
+                loadSubcategory(e.target.value);
+            }
+        });
+
+        function clearSubcategories() {
+            $subcategoryEl.empty();
+            $subcategoryEl.append($("<option></option>").attr("value", "").text("----------"));
```
[the remark this answers is not in the document]
Comment (RomanMelnyk113) [reply wazimap_ng/general/static/js/profile-indicator-admin.js]: fixed this in a bit different way, please take a look at the latest changes.

File: wazimap_ng/general/static/js/profile-indicator-admin.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+            console.log("selectedProfile", selectedProfile)
+            if (selectedProfile) {
+                var url = `/api/v1/profiles/${selectedProfile}/subcategories/`;
+                $.ajax({
+                    url: url,
+                    success: function (data) {
+                        // clear it here to avoid "select" input flickering which appears during AJAX call
+                        clearSubcategories();
+                        $.each(data, function (key, value) {
+                            $subcategoryEl.append($("<option></option>").attr("value", value.id).text(value.name));
```
[the remark this answers is not in the document]
Comment (RomanMelnyk113) [reply wazimap_ng/general/static/js/profile-indicator-admin.js]: see comment above

**Review by milafrerichs — APPROVED** [review]
